### PR TITLE
Adding FS TC_83591419

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding FS TC_83591419

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GESAUF/

Updated Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-E043SK/

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by neha-gangadhar** [comment]
@AmarnatReddy Is the PR intended only for the functions? or would you be adding the test case? Confirming for review purposes.

**Comment by neha-gangadhar** [comment]
@AmarnatReddy Please add the suite to quincy release as well. Also, are we targeting to cover all test scenarios documented in the Test gaps doc in the same PR? or is the PR only for basic workflows? and other test scenarios will be enhanced later?

**Comment by AmarnatReddy** [comment]
> @AmarnatReddy Please add the suite to quincy release as well. Also, are we targeting to cover all test scenarios documented in the Test gaps doc in the same PR? or is the PR only for basic workflows? and other test scenarios will be enhanced later?

Added Testcase Quincy suite. We are covering all the scenarios listed as part of test gap document Neha.

File: tests/cephfs/clients/validate_root_squash.py [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+            mon_caps="allow *",
+            mds_caps=f"allow * fsname={fs_list[0]} root_squash",
+            osd_caps="allow *",
+        )
+        fs_util.create_ceph_client(
+            client,
+            "client_3",
+            mon_caps="allow *",
+            # mds_caps=f"allow rw fsname={fs_list[0]} root_squash,allow * path=/,allow *
+            # fsname={fs_list[1]} root_squash path=/",
```
Comment (sumabai) [inline tests/cephfs/clients/validate_root_squash.py:100]: Are the commented lines going to be retained during merge? If yes, please add the reason, else please remove it.

File: tests/cephfs/clients/validate_root_squash.py [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+                path="/mnt/client_root_squash_Scenario_5_1/",
+                file_name="test_scenario_5_1",
+                allowed=False,
+            )
+            validate_file_dir(
+                cl,
+                path="/mnt/client_root_squash_Scenario_5_1/volumes/subvol_group_2/subvol_4/",
+                file_name="test_scenario_5_1",
+            )
+        log.info("Test 6: Update client caps and validate it rfelects on the mounts")
```
Comment (sumabai) [inline tests/cephfs/clients/validate_root_squash.py:380]: typo for 'reflects'

File: tests/cephfs/clients/validate_root_squash.py (hunk 1 above)
Comment (AmarnatReddy) [reply tests/cephfs/clients/validate_root_squash.py:133]: Added the comment suma

File: tests/cephfs/clients/validate_root_squash.py (hunk 2 above)
Comment (AmarnatReddy) [reply tests/cephfs/clients/validate_root_squash.py]: updated

**Review by neha-gangadhar — COMMENTED** [review]
lgtm

**Review by neha-gangadhar — APPROVED** [review]
Approved

**Pull request merged**